pkgs/dart_mcp/README.md

```diff
@@ -1,5 +1,75 @@
-Experimental Dart SDK for implementing MCP servers and clients.
+A Dart package for making MCP servers and clients.
 
-## Status
+## Implementing Servers
 
-WIP
\ No newline at end of file
+To implement a server, import `package:dart_mcp/server.dart` and extend the
+`MCPServer` class. You must provide the server a communication channel to send
+and receive messages with.
+
+For each specific MCP capability or utility your server supports, there is a
+corresponding mixin that you can use (`ToolsSupport`, `ResourcesSupport`, etc).
+
+Each mixin has doc comments explaining how to use it - some may require you to
+provide implementations of methods, while others may just expose new methods
+that you can call.
+
+Before attempting to call methods on the client, you should first wait for the
+`initialized` future and then check the capabilities of the client by reading
```

Comment:
can you add the class name here? `SomeClass.initialized`? same for `clientCapabilities`